### DISCUSSION TITLE
Injecting css without duplicates

### DISCUSSION THREAD

**Discussion by ELowry** [ChristopherKlay/StadiaEnhanced#61] [Ideas & Requests]
Here's a quick go at trying to keep the css injection clean:

```javascript
function enhanced_addGlobalStyle(id, css) {
    var head = document.getElementsByTagName('head')[0];
	if (!head) {
		return;
	}
	var style = document.GetElementById(id),
		isnew = style === null;
	if (isnew)
	{
		style = document.createElement('style');
		style.type = 'text/css';
		style.id = id;
	}
	style.innerHTML = css;
	if (isnew)
	{
		head.appendChild(style);
	}
}
```

You'll need to include an `id` for each separate css "block" you use, but it should otherwise be pretty transparent.

I haven't tested it but I am about 99% certain this works as intended.

**Comment by ChristopherKlay** [comment]
```Javascript
function enhanced_injectStyle(content, id) {
    head = document.getElementsByTagName('head')[0];
    if (!head) {
        return;
    }
    if (content) {
        if (document.getElementById(id)) {
            var el = document.getElementById(id);
        } else {
            var el = document.createElement('style');
        }
        el.type = 'text/css';
        el.id = id;
        el.innerHTML = content;
        head.appendChild(el);
    } else {
        if (document.getElementById(id)) {
            var el = document.getElementById(id);
            el.parentNode.removeChild(el);
        }
    }
}
```
Results in a clean CSS overview;
![](https://i.imgur.com/rhGo79n.png)
Slightly more complex, but it allows to remove the style element as well, instead of being forced to set default values, if you want to revert a change. Especially useful in this case because default values might change on Stadias end and this would stop the extension from setting wrong defaults.

Any issues you see with this? Otherwise i'll implement it in `1.8.2` (:

**Reply by ELowry** [reply]
Looking good =)
